File: python/test/usage_example.py
import sys, os
sys.path.insert(0, os.path.dirname(__file__) + "/..")

import inhumate_rti_client as RTI
import time

rti = RTI.Client(application="python_usage_example")

def on_control(message: RTI.proto.RuntimeControl):
    print("Received a control message:", message.load_scenario.name)

rti.subscribe("control", RTI.proto.RuntimeControl, on_control)

def on_text(text: str):
    print(f"Received text: {text}")

rti.subscribe_text("text", on_text)

def on_connect():
    print("Connected")

rti.on("connect", on_connect)

def on_disconnect():
    print("Disconnected")

rti.on("disconnect", on_disconnect)

def on_error(type, message, exception):
    print(f"Error: {type}: {message} {exception}")

rti.on("error", on_error)

try:
    time.sleep(1)
    rti.state = RTI.proto.RuntimeState.RUNNING
    message = RTI.proto.RuntimeControl()
    message.load_scenario.name = "python_test"
    rti.publish("control", message)
    print("Published a control message")
    time.sleep(1)
    input("Press enter or ctrl-c to quit...")
except KeyboardInterrupt:
    pass
rti.disconnect()

File: js/test/usage_example.ts
import * as RTI from "../src/index.js"
import os from "os"

const rti = new RTI.Client({ application: "typescript test", user: "foo", host: os.hostname() })

rti.on("connect", () => {
    console.log("connected!")

    const sub = rti.subscribeText("text", (message: string) => {
        console.log(`received text: ${message}`)
    })
    const sub2 = rti.subscribeText("text", (message: string) => {
        console.log(`received text 2: ${message}`)
    })
    setTimeout(() => {
        rti.publishText("text", "foobar")
    }, 500)
    setTimeout(() => {
        rti.unsubscribe(sub2)
    }, 750)
    setTimeout(() => {
        rti.publishText("text", "only 1 should receive this")
    }, 1000)
    setTimeout(() => {
        rti.unsubscribe(sub)
    }, 1500)
    setTimeout(() => {
        rti.publishText("text", "should not receive this at all")
    }, 2000)

    rti.subscribe(RTI.channel.control, RTI.proto.RuntimeControl, (message: RTI.proto.RuntimeControl) => {
        console.log("received control message", JSON.stringify(RTI.proto.RuntimeControl.toJSON(message)))
    })

    setTimeout(() => {
        console.log("publishing control message")
        //rti.publishBytes(RTI.channel.control, RTI.proto.RuntimeControl.encode({ loadScenario: { name: "foo", parameterValues: {} } }))
        rti.publish(RTI.channel.control, RTI.proto.RuntimeControl, { loadScenario: { name: "foo", parameterValues: {} } })
    }, 1000)

    setTimeout(() => {
        console.log("end")
        rti.disconnect()
    }, 3000)
})

rti.on("disconnect", () => {
    console.log("disconnected")
    process.exit(0)
})

rti.on("error", (message: any) => {
    console.error(`Error: ${message}`)
})
